src/hotspot/share/code/codeCache.cpp

```diff
@@ -310,9 +310,22 @@ void CodeCache::initialize_heaps() {
   FLAG_SET_ERGO(ProfiledCodeHeapSize, profiled_size);
   FLAG_SET_ERGO(NonProfiledCodeHeapSize, non_profiled_size);
 
+  const size_t ps = page_size(false, 8);
+  // Print warning if using large pages but not able to use the size given
+  if (UseLargePages && ps < LargePageSizeInBytes) {
+      char msg[256];
+      jio_snprintf(msg, sizeof(msg),
+                   "Failed to reserve large page memory for code cache (" SIZE_FORMAT "%s). "
+                   "Reverting to smaller page size (" SIZE_FORMAT "%s).",
+                   byte_size_in_exact_unit(LargePageSizeInBytes), exact_unit_for_byte_size(LargePageSizeInBytes),
+                   byte_size_in_exact_unit(ps), exact_unit_for_byte_size(ps));
```

Comment:
You could use EXACTFMT and EXACTFMTARGS to shorten this, see globalDefinitions.hpp